### PR TITLE
`cylc lint`: S015/U017 line continuation chars

### PR DESCRIPTION
**Pull request by wxtim**
Replaces https://github.com/cylc/cylc-flow/pull/6459 which blocks sync PRs

Otherwise the same diff.

Seealso @MetRonnie's Comment on https://github.com/cylc/cylc-flow/pull/6459#issuecomment-2497686715

Probably OK to be a single review since it's a copy of an already accepted PR.

### CODE REVIEW

File: changes.d/6459.feat.md
Comment (MetRonnie) [inline changes.d/6459.feat.md:1]: 6493

Comment (oliver-sanders) [reply changes.d/6459.feat.md:1]: It's ok to reference the issue rather than the PR right?

Comment (MetRonnie) [reply changes.d/6459.feat.md:1]: Might be better in this case

File: cylc/flow/scripts/lint.py [hunk 1]
```diff
@@ -617,7 +617,13 @@ def list_wrapper(line: str, check: Callable) -> Optional[Dict[str, str]]:
             for job_runner, directive in WALLCLOCK_DIRECTIVES.items()
         )),
         FUNCTION: check_wallclock_directives,
-    }
+    },
+    'S015': {
+        'short': (
+            '`=>` is a line continuation without `\\`.'
```
Comment (hjoliver) [inline cylc/flow/scripts/lint.py:623]:
```suggestion
            '`=>` implies line continuation without `\\`.'
```
(Pretty sure you accepted this change on the original!)

**Review by hjoliver — CHANGES_REQUESTED** [review]

**Review by hjoliver — APPROVED** [review]
All good 🎉

File: changes.d/6459.feat.md
Comment (MetRonnie) [reply changes.d/6459.feat.md:1]: #6497

**Pull request merged**